internal/testing/e2e/backfill_test.go

```diff
@@ -66,3 +68,134 @@ func TestCreateGetBackfill(t *testing.T) {
 	require.Equal(t, codes.NotFound.String(), status.Convert(err).Code().String())
 	require.Nil(t, actual)
 }
+
+func TestProposedBackfillCreate(t *testing.T) {
+	ctx := context.Background()
+	om := newOM(t)
+
+	t1, err := om.Frontend().CreateTicket(ctx, &pb.CreateTicketRequest{
+		Ticket: &pb.Ticket{
+			SearchFields: &pb.SearchFields{
+				StringArgs: map[string]string{
+					"field": "value",
+				},
+			},
+		},
+	})
+	require.Nil(t, err)
+
+	b := &pb.Backfill{
+		SearchFields: &pb.SearchFields{
+			StringArgs: map[string]string{
+				"field": "value",
+			},
+		},
+	}
+	m := &pb.Match{
+		MatchId:  "1",
+		Tickets:  []*pb.Ticket{t1},
+		Backfill: b,
+	}
+
+	om.SetMMF(func(ctx context.Context, profile *pb.MatchProfile, out chan<- *pb.Match) error {
+		out <- m
+		return nil
+	})
+
+	om.SetEvaluator(func(ctx context.Context, in <-chan *pb.Match, out chan<- string) error {
+		p, ok := <-in
+		require.True(t, ok)
+
+		out <- p.MatchId
+		return nil
+	})
+
+	stream, err := om.Backend().FetchMatches(ctx, &pb.FetchMatchesRequest{
+		Config:  om.MMFConfigGRPC(),
+		Profile: &pb.MatchProfile{},
+	})
+	require.Nil(t, err)
+	resp, err := stream.Recv()
+	require.Nil(t, err)
+
+	actual, err := om.Frontend().GetBackfill(ctx, &pb.GetBackfillRequest{BackfillId: resp.Match.Backfill.Id})
+	require.Nil(t, err)
+	require.NotNil(t, actual)
+
+	b.Id = actual.Id
+	b.CreateTime = actual.CreateTime
+	require.True(t, proto.Equal(b, actual))
+	
+	client, err := om.Query().QueryTickets(ctx, &pb.QueryTicketsRequest{Pool: &pb.Pool{
+		StringEqualsFilters: []*pb.StringEqualsFilter{{StringArg: "field", Value: "value"}},
+	}})
+
+	require.Nil(t, err)
+	_, err = client.Recv()
+	require.Equal(t, io.EOF, err)
+}
+
+func TestProposedBackfillUpdate(t *testing.T) {
+	ctx := context.Background()
+	om := newOM(t)
+	t1, err := om.Frontend().CreateTicket(ctx, &pb.CreateTicketRequest{
+		Ticket: &pb.Ticket{
+			SearchFields: &pb.SearchFields{
+				StringArgs: map[string]string{
+					"field": "value",
+				},
+			},
+		},
+	})
+	require.Nil(t, err)
+
+	b, err := om.Frontend().CreateBackfill(ctx, &pb.CreateBackfillRequest{Backfill: &pb.Backfill{
+		SearchFields: &pb.SearchFields{
+			StringArgs: map[string]string{
+				"field": "value",
+			},
+		},
+	}})
+	require.Nil(t, err)
+
+	m := &pb.Match{
+		MatchId:  "1",
+		Tickets:  []*pb.Ticket{t1},
+		Backfill: b,
```

Comment:
Please test the mmf changing a value on the backfill.

Reply:
I will check changes to search fields and extensions fields will be applied.